pkg/services/ngalert/notifier/channels/template_data.go

```diff
@@ -89,6 +89,34 @@ func extendAlert(alert template.Alert, externalURL string, logger log.Logger) *E
 			u.RawQuery = "viewPanel=" + panelId
 			extended.PanelURL = u.String()
 		}
+
+		generatorUrl, gErr := url.Parse(extended.GeneratorURL)
+		if gErr != nil {
+			logger.Debug("failed to parse generator URL while extending template data", "url", extended.GeneratorURL, "err", gErr.Error())
+			return extended
+		}
+
+		dashboardUrl, dErr := url.Parse(extended.DashboardURL)
+		if dErr != nil {
+			logger.Debug("failed to parse dashboard URL while extending template data", "url", extended.DashboardURL, "err", dErr.Error())
+			return extended
+		}
+
+		orgId := alert.Annotations[ngmodels.OrgIDAnnotation]
+		if len(orgId) > 0 {
+			orgIdParam := "orgId=" + orgId
+
+			dashboardUrl.RawQuery = orgIdParam
```

Comment:
Why use RawQuery here and Query.Add three lines down? Let's be consistent.

Reply:
Have extracted the logic of appending the query param to a private function to keep the logic consistent.